src/engine/cachingreader.cpp

```diff
@@ -218,24 +242,23 @@ void CachingReader::process() {
                 // Discard chunks that don't carry any data
                 freeChunk(pChunk);
             }
-        }
-        if (update.status == TRACK_NOT_LOADED) {
-            m_readerStatus = update.status;
-        } else if (update.status == TRACK_LOADED) {
-            m_readerStatus = update.status;
-            // Reset the max. readable frame index
-            m_readableFrameIndexRange = update.readableFrameIndexRange();
-            // Free all chunks with sample data from a previous track
-            freeAllChunks();
-        }
-        if (m_readerStatus == TRACK_LOADED) {
-            // Adjust the readable frame index range after loading or reading
-            m_readableFrameIndexRange = intersect(
-                    m_readableFrameIndexRange,
-                    update.readableFrameIndexRange());
+            // Adjust the readable frame index range (if available)
+            if (update.status != CHUNK_READ_DISCARDED) {
+                m_readableFrameIndexRange = intersect(
+                        m_readableFrameIndexRange,
+                        update.readableFrameIndexRange());
+            }
         } else {
```

Comment:
Are we sure that we hit always the else branch? We need to receive status update without a chunk. 
I think it would be better to have that decoupled

Reply:
Yes. This is caused by the coupled design. But I won't change that. The debug assertions clearly indicate that there are two kinds of updates:
- Read result with a chunk
- State change track loaded/unloaded without a chunk